server/boot/07-weather.js

```diff
@@ -0,0 +1,212 @@
+  },
+  {
+    'name': 'Redskins',
+    'location': {'lat': 38.907726, 'lon': -76.864278},
+    'dome': false
+  },
+  {
+    'name': 'Patriots',
+    'location': {'lat': 42.090978, 'lon': -71.263950},
+    'dome': false
+  },
+  {
+    'name': 'Lions',
+    'location': {'lat': 42.339991, 'lon': -83.045227},
+    'dome': true
+  },
+  {
+    'name': 'Vikings',
+    'location': {'lat': 44.973590, 'lon': -93.257136},
+    'dome': true
+  },
+  {
+    'name': 'Chiefs',
+    'location': {'lat': 39.048935, 'lon': -94.483322},
+    'dome': false
+  },
+  {
+    'name': 'Giants',
+    'location': {'lat': 40.812840, 'lon': -74.073952},
+    'dome': false
+  },
+  {
+    'name': 'Jets',
+    'location': {'lat': 40.812840, 'lon': -74.073952},
+    'dome': false
+  },
+  {
+    'name': 'Bills',
+    'location': {'lat': 42.773713, 'lon': -78.786598},
+    'dome': false
+  },
+  {
+    'name': 'Panthers',
+    'location': {'lat': 35.225147, 'lon': -80.851775},
+    'dome': false
+  },
+  {
+    'name': 'Bengals',
+    'location': {'lat': 39.095464, 'lon': -84.516371},
+    'dome': false
+  },
+  {
+    'name': 'Browns',
+    'location': {'lat': 41.505981, 'lon': -81.698904},
+    'dome': false
+  },
+  {
+    'name': 'Eagles',
+    'location': {'lat': 39.900841, 'lon': -75.166573},
+    'dome': false
+  },
+  {
+    'name': 'Steelers',
+    'location': {'lat': 40.446830, 'lon': -80.015578},
+    'dome': false
+  },
+  {
+    'name': 'Titans',
+    'location': {'lat': 36.166518, 'lon': -86.771048},
+    'dome': false
+  },
+  {
+    'name': 'Cowboys',
+    'location': {'lat': 32.747167, 'lon': -97.093754},
+    'dome': true
+  },
+  {
+    'name': 'Texans',
+    'location': {'lat': 29.684722, 'lon': -95.410150},
+    'dome': true
+  },
+  {
+    'name': 'Seahawks',
+    'location': {'lat': 47.595239, 'lon': -122.331350},
+    'dome': false
+  },
+  {
+    'name': 'Packers',
+    'location': {'lat': 44.501253, 'lon': -88.061656},
+    'dome': false
+  }
+];
+
+module.exports = function(app) {
+  console.log('Starting boot script 7');
+  var Weather = app.models.Weather;
+  var apiKey = '0611c31eb00b48fa862ed5f95251fab3';
+  cron.schedule('0 0 * * * *', function() {
+    // Get current season and week in NFL
+      console.log('Fetching weather data');
+    team_map.forEach(function (team) {
+      console.log("processing weather for ", team.name);
+      request.get('http://api.openweathermap.org/data/2.5/weather')
+      .query({ lat: team.location.lat, lon: team.location.lon, appid: apiKey }) // query string
+      .end(function(err, res){
+        var condition = {
+          temp: res.body.main.temp - 273.15,
+          name: res.body.name,
+          description: res.body.weather[0].description
+        };
+        console.log('weather response for ', team.name);
+        var today_date = new Date().toJSON().slice(0,10).replace(/-/g,'-');
+        Weather.findOne({where: {team: team.name, created_date: today_date}}).then(function (weather) {
```

Comment:
If the names of the teams are a unique key, then you could use `Weather.upsert` where it will either update or create a new model instance: [PersistedModel.upsert](http://apidocs.loopback.io/loopback/#persistedmodel-upsert)

Reply:
I need the name and the created_date to be unique together

Reply:
You can make the fields unique in your weather.json file like this:
```
"team": {
  "type": "string",
  "required": true,
  "index": {
    "unique": true
  }
}
``` 
Then `Weather.upsert` would work.